src/lib/comms/sol-socket-dtls-impl-tinydtls.c

```diff
@@ -907,3 +840,35 @@ sol_socket_dtls_prf_keyblock(struct sol_socket *s,
 
     return 0;
 }
+
+int
+sol_socket_dtls_set_credentials_callbacks(struct sol_socket *s,
+    const struct sol_socket_dtls_credential_cb *cb)
+{
+    struct sol_socket_dtls *socket = (struct sol_socket_dtls *)s;
+
+    SOL_INT_CHECK(socket->dtls_magic, != dtls_magic, -EINVAL);
+
+    socket->credentials = cb;
+
+    return 0;
+}
+
+int
+sol_socket_dtls_close(struct sol_socket *s, struct sol_network_link_addr *addr)
```

Comment:
Or, better yet, just close the socket and open a new one. It's cheap, and this will help destroy any data that could be potentially left over from the time the socket was being used for the initial exchange.